torch/_dynamo/variables/torch.py

```diff
@@ -670,7 +670,12 @@ def handle_from_local(self, tx: "InstructionTranslator", *args, **kwargs):
                 # rewrite non-primitive args/kwargs to be included in the on-the-fly prim function
                 # and rewrite args to have only proxyable args, then insert call_function
                 args_as_value = [x.as_python_constant() for x in args[1:]]
-                kwargs_as_value = {k: v.as_python_constant() for k, v in kwargs.items()}
+                kwargs_as_value = {
+                    k: v.as_proxy()
+                    if k in ["shape", "stride"]
+                    else v.as_python_constant()
+                    for k, v in kwargs.items()
+                }
 
                 def fn_with_prim_types(x):
```

Comment:
hmm, I also expected you to have to do something like in order for the dynamic shapes to actually flow through the graph:
```
if has_symints(kwargs['shape']) and has_symints(kwargs['stride']):
    def fn_with_prim_type(x, shape, stride):
        return self.value(x, *args_as_value, **kwargs_as_value, shape=shape, stride=stride)
elif has_symints(kwargs['shape']):
    def fn_with_prim_type(x, shape):
        ...
elif has_symints(kwargs['stride']):
      def fn_with_prim_type(x, stride):
         ....
else:
    def fn_with_prim_types(x):
        ....
```

we have a bunch of unit tests that test the `from_local()` constructor in-graph with compile [here](https://github.com/pytorch/pytorch/blob/main/test/distributed/_tensor/test_dtensor_compile.py#L264) - do you mind adding a few similar tests like:

(1) `from_local()` in graph with both sizes and strides past in, with all dims dynamic (`compile(dynamic=True)`)
(2) same but only sizes are dynamic and not strides (e.g. a 1-d tensor with dynamic size, but stride of 1)
(3) same as (1), but where the sizes and strides have a mix of static and dynamic dims, e.g. sizes=(s1, s0, 2), strides=(s0*2, 2, 1)

Reply:
learned from Brian:

for a test where some sizes are dynamic and others are not, you can use mark_dynamic:
```
inp_tensor = torch.randn(8, 4)
# marks the first dim as a dynamic, but second will remain static
torch._dynamo.mark_dynamic(inp_tensor, 0)
# pass it into a compiled region
out = f(inp_tensor)
```
(so you're not actually passing SymInts into the compiled region - you're just annotating your runtime sizes to tell dynamo which input sizes should be converted to SymInts in the compiler)